task_manager_app/views.py

```diff
@@ -0,0 +1,111 @@
+from django.utils import timezone
+from rest_framework import status
+from rest_framework.decorators import api_view
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from task_manager_app.models import Task, SubTask
+from task_manager_app.serializer import TaskSerializer, CreateTaskSerializer, TaskDetailSerializer, SubTaskSerializer, \
+    CreateSubTaskSerializer
+
+
+
+class SubTaskListCreateView(APIView):
+    def get(self, request):
+        subtasks = SubTask.objects.all()
+        serializer = SubTaskSerializer(subtasks, many=True)
+        return Response(serializer.data, status=status.HTTP_200_OK)
+
+    def post(self, request):
+        task = CreateSubTaskSerializer(data=request.data)
+        if task:
+            if task.is_valid():
+                task.save()
+            return Response(status=status.HTTP_204_NO_CONTENT)
+        return Response(status=status.HTTP_400_BAD_REQUEST)
+
+
+
+
+class SubTaskDetailUpdateDeleteView(APIView):
+    def get_subtask(self, pk):
+        try:
+            return SubTask.objects.get(pk=pk)
+        except SubTask.DoesNotExist:
+            return None
+
+    def patch(self, request, pk):
+        subtask = self.get_subtask(pk)
+        if not subtask:
+            return Response("{'error' 'Subtask not found'}",status=status.HTTP_404_NOT_FOUND)
```

Comment:
Тут можно сделать сразу вызов соответствующего ответа. Согда кот в двух методах станет заметно проще
```suggestion
    def get_subtask(self, pk):
        return get_object_or_404(SubTask, pk=pk)

    def patch(self, request, pk):
        subtask = self.get_subtask(pk)
```